fetch.bs

```diff
@@ -255,6 +258,29 @@ given a <a for=/>fetch timing info</a> <var>timingInfo</var>, return a new
  <var>storedTimingInfo</var>'s <a for="fetch timing info">decoded body size</a>.
 </ol>
 
+<p>To <dfn for="fetch params">abort</dfn> a <a for=/>fetch params</a> <var>fetchParams</var>,
+set <var>fetchParams</var>'s <a for="fetch params">fetch state</a> to "<code>aborted</code>".
+
+<p>To <dfn for="fetch params">terminate</dfn> a <a for=/>fetch params</a> <var>fetchParams</var>,
+set <var>fetchParams</var>'s <a for="fetch params">fetch state</a> to "<code>terminated</code>".
+
+<p>A <a for=/>fetch params</a> <var>fetchParams</var> is <dfn for="fetch params">canceled</dfn> if
+its <a for="fetch params">fetch state</a> is "<code>aborted</code>" or "<code>terminated</code>".
+
+<p>A <dfn export>fetch controller</dfn> is an object used to enable clients of an ongoing fetch
+to perform certain actions in the context of that fetch instance.
```

Comment:
It seems a bit weird for state to be terminated. But I do like this idea. Can it instead be the controller that holds the state? And fetch params / fetch record have a reference to the controller?

And if we need to expose more things on the controller in due course, we could do that this way? And only expose it at the appropriate time, etc.

That does seem somewhat cleaner than having an internal slot and hiding it by not exporting it.

Reply:
> It seems a bit weird for state to be terminated. But I do like this idea. Can it instead be the controller that holds the state? And fetch params / fetch record have a reference to the controller?
> 
> 
> 
> And if we need to expose more things on the controller in due course, we could do that this way? And only expose it at the appropriate time, etc.
> 
> 
> 
> That does seem somewhat cleaner than having an internal slot and hiding it by not exporting it.

Sure, I will revise 
It might feel clunky with `timing info` but we can cross the bridge when we get there

Reply:
I suspect that might be something we need to keep on both. But once it's put on controller it isn't modified again, or some such.

Reply:
Gotcha 
Let me put out the PR for this one and I will tinker with how this works for timing

Reply:
@annevk 

> It seems a bit weird for state to be terminated. But I do like this idea. Can it instead be the controller that holds the state? And fetch params / fetch record have a reference to the controller?

fwiw, that's what I was trying to say, although I then renamed the controller since it was just a state holder.

Reply:
It's a state holder now, but when I put "finalize and report timing" there it will do a bit more.
The controller is what allows you to terminate/finalize the fetch after it's started, the fact that it holds a state is less relevant since that state is not readable from the outside, only writable with terminate/abort/finalize.